### DISCUSSION TITLE
Difficulty splitting field with \n in PnP Search Results v4

### DISCUSSION THREAD

**Discussion by ansbbell** [microsoft-search/pnp-modern-search#3803] [Q&A]
I have searched for "split" in every discussion here, but did not get anywhere (am a novice), and it would be great if anyone knows the answer.  Some of my code is below to try to help explain the question, but the code itself is working as long as the first split operation uses something other than \n and I don't need help with that, only with the split operation.

I have a library column with multiple lines of text.  Each line is formatted as [Title]url, ex:

[Link 1]https://www.google.com
[Link 2]https://www.github.com
etc.

The output from the column is assigned to a slot named "Links" in my PnP Search Results web part.  I was trying to use the following in my custom results template to display these

```
{{#if (slot item @root.slots.Links)}}
   {{#each (split (slot item @root.slots.Links) "\n") as |disp| }}
      {{#with (split (disp) "]") }}
        <a class="example-displayLink custom-link" href="{{[1]}}" target="_blank" data-interception="off" >{{remove [0] "["}}</a>
      {{/with}}
   {{/each}}
{{/if}}
```
but I could not get the split operation to work with "\n" and had to resort to adding a semi-colon at the end of each line in the library column and splitting with that

`{{#each (split (slot item @root.slots.Links) ";") as |disp| }}`

The ending semi-colon can be removed in the anchor element and it works great.  However, while it is relatively easy to ask people to paste a URL in SharePoint when they upload something, it is also too easy for them to forget to put a semi-colon at the end.  So is there a actually a way to split with "\n"?  

In case it is relevant, my org is using pnp-modern-search-parts-v4 version 4.9.3.0. but I probably do not have enough clout to ask them to update.

**Comment by markder** [comment]
Try using `"\\n"` instead of `"\n".`

I had a similar issue where I was using `'\'` as the deliminator in a split. I had to escape it by using `'\\'.`

**Reply by ansbbell** [reply]
Thanks for the suggestion @markder .  I just tried this, and also a few other permutations for escaping the \ (suggested by Claude Sonnet, understand what they are but not whether they make sense;  easy enough to see what happens) - `"\r\n" "\\r\\n" "\\s*\\n\\s*"`, even "<br\>", but the only thing that works for me is adding a semi-colon and using that to split.  Just for reference, I will put a picture of the debug output when \n is used.  Field in question is LinksOWSMTXT, just above the lines that have been blurred.  This is weird but not a big deal since there is a workaround.


![picture-of-debug-properties-without-semi-colon](https://github.com/microsoft-search/pnp-modern-search/assets/172845014/9ce4713d-efe8-4097-ae35-ddee2243185b)

**Comment by markder** [comment]
That is weird. The only other guess I have is maybe using single quotes instead of double. I didn't think this would make a difference because you used double quotes successfully with the semi-colon but it did help in my scenario.

**Reply by ansbbell** [reply]
Thanks for the additional suggestion.  For reference of anyone reading this in the future, I did some more tests, attempting to split with `'\n', '\\n', and ';'`  There was no change in either result with newline, but interestingly, splitting with the semi-colon works with single quotes as well as double quotes.  An example I found before starting on this used double quotes, so I was doing that.  Not sure what the formal requirements might be.
